src/dftbp/dftbplus/parser.F90

```diff
@@ -7707,50 +7707,104 @@ end function is_numeric
+
+    !> Range separated data structure to fill
+    type(TRangeSepInp), intent(inout), allocatable :: input
+
+    !> List of SK file names to read in for every interaction
+    type(TListCharLc), intent(inout) :: skFiles(:,:)
+
+    !! File name of representative SK-file to read
+    character(lc) :: fileName
+
+    !! Range-separated extra tag in SK-files, if allocated
+    integer :: rangeSepSkTag
 
-    type(fnode), pointer :: child1, value1, child2, value2, child3
+    !! Range-separated functional type of user input
+    integer :: rangeSepInputTag
+
+    !! Auxiliary node pointers
+    type(fnode), pointer :: hybridChild, hybridValue, screeningChild, screeningValue, child1
+
+    !! True, if RangeSeparated input block present
+    logical :: isHybridInp
+
+    !! True, if range-separated extra tag found in SK-file(s)
+    logical :: isHybridSk
+
+    !! Temporary string buffers
     type(string) :: buffer, modifier
 
-    call getChildValue(node, "RangeSeparated", value1, "None", child=child1)
-    call getNodeName(value1, buffer)
+    @:ASSERT(size(skFiles, dim=1) == size(skFiles, dim=2))
+    @:ASSERT((size(skFiles, dim=1) > 0))
 
-    select case (char(buffer))
+    ! Extracting range-separated tag from first SK-file only is a workaround and assumes that a set
+    ! of given SK-files uses the same parameters (which should always be the case)!
+    call get(skFiles(1, 1), fileName, 1)
 
-    case ("none")
-      continue
+    ! Check if SK-files contain extra tag for range-separated functionals
+    call inquireRangeSepTag(fileName, rangeSepSkTag)
+    isHybridSk = rangeSepSkTag /= rangeSepFunc%none
 
-    case ("lc")
-      allocate(input)
-      call getChildValue(value1, "Screening", value2, "Thresholded", child=child2)
-      call getNodeName(value2, buffer)
-      select case(char(buffer))
-      case ("neighbourbased")
-        input%rangeSepAlg = rangeSepTypes%neighbour
-        call getChildValue(value2, "CutoffReduction", input%cutoffRed, 0.0_dp,&
-            & modifier=modifier, child=child3)
-        call convertUnitHsd(char(modifier), lengthUnits, child3, input%cutoffRed)
-      case ("thresholded")
-        input%rangeSepAlg = rangeSepTypes%threshold
-        call getChildValue(value2, "Threshold", input%screeningThreshold, 1e-6_dp)
-        call getChildValue(value2, "CutoffReduction", input%cutoffRed, 0.0_dp,&
-            & modifier=modifier, child=child3)
-        call convertUnitHsd(char(modifier), lengthUnits, child3, input%cutoffRed)
-      case ("matrixbased")
-        input%rangeSepAlg = rangeSepTypes%matrixBased
-        ! In this case, CutoffRedunction is not used so it should be set to zero.
-        input%cutoffRed = 0.0_dp
+    call getChild(node, "RangeSeparated", child=hybridChild, requested=.false.)
+    isHybridInp = associated(hybridChild)
+
+    if (isHybridInp .and. .not. isHybridSk) then
+      call error("RangeSeparated input block present, but SK-files seem to be (semi-)local.")
```

Comment:
Could you print the name of the problem file in the error?

Reply:
Sure, will do after lunch ;)

Reply:
done